### PR TITLE
[type:refactor] optimize dubbo ReferenceConfig destroy

### PR DESCRIPTION
**Pull request by zhaoguhong**
optimize dubbo ReferenceConfig destroy

### CODE REVIEW

File: shenyu-plugin/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApplicationConfigCache.java [hunk 1]
```diff
@@ -56,16 +55,7 @@
...
-                        Class<?> cz = config.getClass();
-                        Field field = cz.getDeclaredField("ref");
-                        field.setAccessible(true);
-                        // After the configuration change, Dubbo destroys the instance, but does not empty it. If it is not handled,
-                        // it will get NULL when reinitializing and cause a NULL pointer problem.
-                        field.set(config, null);
-                    } catch (NoSuchFieldException | IllegalAccessException e) {
-                        LOG.error("modify ref have exception", e);
-                    }
+                    config.destroy();
```
Comment (yu199195) [inline shenyu-plugin/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApplicationConfigCache.java:58]: config.destroy();  maybe have some error

Comment (zhaoguhong) [reply shenyu-plugin/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApplicationConfigCache.java:58]: Can you tell me what's error

**Comment by tydhot** [comment]
plz check the conficts

**Comment by yu199195** [comment]
plz check the conficts

**Comment by KevinClair** [comment]
Refer to #2636

**Pull request closed without merging**